src/main/kotlin/org/eclipse/uprotocol/rpc/RpcMapper.kt

```diff
@@ -30,129 +30,72 @@ import com.google.protobuf.Message
 import java.util.concurrent.CompletionException
-import java.util.concurrent.CompletionStage
 
-/**
- * RPC Wrapper is an interface that provides static methods to be able to wrap an RPC request with
- * an RPC Response (uP-L2). APIs that return Message assumes that the payload is protobuf serialized
- * com.google.protobuf.Any (UPayloadFormat.PROTOBUF) and will barf if anything else is passed
- */
-interface RpcMapper {
-    companion object {
-        /**
-         * Map a response of CompletionStage&lt;UPayload&gt; from Link into a CompletionStage containing the declared expected return type of the RPC method or an exception.
-         * @param responseFuture CompletionStage&lt;UPayload&gt; response from uTransport.
-         * @param expectedClazz The class name of the declared expected return type of the RPC method.
-         * @return Returns a CompletionStage containing the declared expected return type of the RPC method or an exception.
-         * @param <T> The declared expected return type of the RPC method.
-        </T> */
-        fun <T : Message> mapResponse(
-            responseFuture: CompletionStage<UPayload>,
-            expectedClazz: Class<T>
-        ): CompletionStage<T> {
-            return responseFuture.handle { payload, exception ->
-                // Unexpected exception
-                if (exception != null) {
-                    throw CompletionException(exception.message, exception)
-                }
-                if (payload == null) {
-                    throw RuntimeException("Server returned a null payload. Expected " + expectedClazz.getName())
-                }
-                val any: Any
-                try {
-                    any = Any.parseFrom(payload.value)
 
-                    // Expected type
-                    if (any.`is`(expectedClazz)) {
-                        return@handle unpackPayload(any, expectedClazz)
-                    }
-                } catch (e: InvalidProtocolBufferException) {
-                    throw RuntimeException(String.format("%s [%s]", e.message, UStatus::class.java.getName()), e)
-                }
-                // Some other type instead of the expected one
-                throw RuntimeException(
-                    String.format(
-                        "Unknown payload type [%s]. Expected [%s]",
-                        any.typeUrl,
-                        expectedClazz.getName()
-                    )
-                )
-            }
+/**
+ * Map a response of Result&lt;UPayload&gt; from Link into a Result containing the declared expected return type of the RPC method or an exception.
+ * @param expectedClazz The class name of the declared expected return type of the RPC method.
+ * @return Returns Result containing the declared expected return type of the RPC method.
+ * @param <T> The declared expected return type of the RPC method.
+</T> */
+fun <T : Message> Result<UPayload>.mapResponse(expectedClazz: Class<T>): Result<T> {
```

Comment:
Let's consider a reified type here. In kotlin, such type of mapping is not required with inline reified T. 
So ideally, we can probably have a `inline fun <reified T: Message> Flow<UPayload>.toMessage(): Flow<T>`